collectors/lib/samza_metric_reporter.py

```diff
@@ -113,6 +115,19 @@ def print_jvm_and_container_metric(self, metric_type, metric_name, ts, value, ta
     def print_consumer_lag(self, ts, value, tags):
         if self.is_number(value):
             print ("%s %d %s %s" % (self.SAMZA_CONSUMER_LAG_METRIC_NAME, ts, value, self.to_tsdb_tag_str(tags)))
+            # Send to DataDog
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) # UDP
```

Comment:
Is there any caching or pooling of this socket? do you need to close the connection? Seems like it would be more efficient to have a single socket initialized globally, send all your messages and then close the connection.

Reply:
It's UDP (connectionless) so it's not needed (fire and forget it).